### PR TITLE
fix: Add information about local development for RWD projects

### PR DESCRIPTION
**Pull request by lasjorg**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

As per https://github.com/freeCodeCamp/testable-projects-fcc/pull/1092

This PR adds a short note to the [Introduction to the Responsive Web Design Projects](https://www.freecodecamp.org/learn/responsive-web-design/responsive-web-design-projects/) page about the need to serve files in order to get accurate test results when doing local development. It includes a list of small development servers.

### CODE REVIEW

**Review by hassaanp — DISMISSED** [review]

File: client/src/pages/learn/responsive-web-design/responsive-web-design-projects/index.md [hunk 1]
```diff
@@ -15,3 +15,5 @@ In this section you get the chance to:
  * Build a Personal Portfolio Webpage
 
 By the end of this, you would have 5 responsive websites under your belt that you can show off to friends, family, employers, etc. Have fun and remember to use the [Read-Search-Ask](https://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck-coding/19514) method if you get stuck.
+
+**Note**: If you work locally on the projects you must use a server in order to get accurate results from the test. There are small development servers like, [Live Server](https://www.npmjs.com/package/live-server) (VS Code [extension](https://ritwickdey.github.io/vscode-live-server/)), [http-server](https://www.npmjs.com/package/http-server) and Python [http.server](https://docs.python.org/3/library/http.server.html) you can use for this.
```
Comment (moT01) [inline client/src/pages/learn/responsive-web-design/responsive-web-design-projects/index.md:19]:
```suggestion
**Note**: If you work locally on the projects you must use a server in order to get accurate results from the test.
```

**Review by moT01 — COMMENTED** [review]
I think we want to avoid external links as much as possible, so I made a suggestion to just have a short note. It is certainly up for discussion.

File: client/src/pages/learn/responsive-web-design/responsive-web-design-projects/index.md (hunk 1 above)
Comment (RandellDawson) [reply client/src/pages/learn/responsive-web-design/responsive-web-design-projects/index.md:19]: @lasjorg If you are ok with @moT01's suggestion, I think we can approve/merge this PR.

**Comment by lasjorg** [comment]
@moT01 Why would external links be a problem in the intro section? Isn't the point of no external links to avoid disturbing the flow when working with challenges?

I think that just saying "use a server" to someone at this stage in the learning process isn't very helpful. If it is a requirement that the page is served for the tests to pass (which it is) surely we should help the camper and provide some simple options for them to use.

**Comment by Sky020** [comment]
> I think that just saying "use a server" to someone at this stage in the learning process isn't very helpful. If it is a requirement that the page is served for the tests to pass (which it is) surely we should help the camper and provide some simple options for them to use.

@lasjorg The way I see it, fCC does not teach campers how to use a local dev environment. So, there are simple options for them to use, in Repl.it, CodePen, etc. If a camper is intent enough on using their own environment, they are probably intent enough to do research into hosting it.

All that said, what about an internal link to a guide/news article with all of these options laid out?

**Comment by RandellDawson** [comment]
> All that said, what about an internal link to a guide/news article with all of these options laid out?

@scissorsneedfoodtoo What do you think about adding such a news article we can link to?

**Comment by lasjorg** [comment]
I still don't understand what the problem is. Can someone please explain to me what the problem is with having external resource links in the intro section?

>If a camper is intent enough on using their own environment, they are probably intent enough to do research into hosting it.

I feel like this response is just passing the ball and I don't know why? What is so wrong with helping the camper by providing some options?

**Comment by raisedadead** [comment]
> I still don't understand what the problem is. Can someone please explain to me what the problem is with having external resource links in the intro section?

Thanks for your query and patience with a response. 

The bottomline boils down to: 

We do not want to and need to endorse any resource/link/site/tool. Obviously there are times when "need to" and make exceptions looking at the PR, I do not think that its a need in this case.

Please note getting stuck, asking queries and then choosing a tool/resource is an active learning exercise.

The reason: 

"How they work" on projects is a recommendation. We do that already with resources that others listed. Working locally is a user choice, its acceptable and supported, but its not our recommendation. 

We do not want users to lose context and scope of the topic at hand, which is the project/challenge. Most of our users are beginners, every extra bit of information is a bias we can avoid.

Finally we also have a technical reason, which is because we do not control external links/resources we should avoid them as much as we can.

Thanks again for your PR. Happy contributing.

**Comment by lasjorg** [comment]
@raisedadead The non-endorsement part and the fact that it is resources out of our control I can understand and that does make sense. Thank you for the explanation.

I still feel like we might be a bit terse in the message of using a server. Maybe we can keep the "There are small development servers you can use for this" part of the original text? Or as suggested link to a news article if possible.

In any case, I will commit the suggested change so it can be merged if no further change is needed.

**Comment by raisedadead** [comment]
I am not sure what "small" means when we talk about development servers.

**Comment by lasjorg** [comment]
> I am not sure what "small" means when we talk about development servers.

I guess by small I mean a simple local HTTP server.

Anyway, it would be nice to have a hits/guide page linked to from the intro page with a few tips and maybe a quick run-down of the testing script.

**Review by RandellDawson — APPROVED** [review]
LGTM

**Comment by raisedadead** [comment]
> I guess by small I mean a simple local HTTP server.


Ah - I see what you mean.

> Anyway, it would be nice to have a hits/guide page linked to from the intro page with a few tips and maybe a quick run-down of the testing script.

Yep, I think that is a doable idea. We chould use a forum post for this.

**Review by raisedadead — APPROVED** [review]

**Comment by RandellDawson** [comment]
> We could use a forum post for this.

Like a Wiki Guide topic?

**Comment by scissorsneedfoodtoo** [comment]
@RandellDawson, I don't think we have a great news article to link to, or at least one with all the different options mentioned in the first commit without going too in depth.

**Comment by RandellDawson** [comment]
@lasjorg If you want to create a new Wiki on the forum, I can transfer the ownership of it over to Camperbot.  We could temporarily keep the wiki hidden until others have had a chance to review it.

**Pull request merged**